src/Microsoft.DotNet.Build.Tasks.Feed/BlobFeed.cs

```diff
@@ -165,63 +165,95 @@ public string CalculateBlobPath(string item, string relativePath)
         public string GeneratePackageServiceIndex(string redirectUrl, IEnumerable<string> values)
         {
             string pathToTempFolder = Path.Combine(IndexDirectory, redirectUrl);
-            if (Directory.Exists(pathToTempFolder))
-            {
-                Directory.Delete(pathToTempFolder, true);
-            }
-            Directory.CreateDirectory(pathToTempFolder);
             string packageIndexJsonLocation = Path.Combine(pathToTempFolder, "index.json");
-            using (var streamWriter = new StreamWriter(File.OpenWrite(packageIndexJsonLocation)))
-            using (JsonTextWriter writer = new JsonTextWriter(streamWriter))
+            int retry = 5;
+            bool isFileCreated = false;
+            while (!isFileCreated && --retry > 0)
             {
-                writer.WriteStartObject();
-                writer.WritePropertyName("versions");
-                writer.WriteStartArray();
-                foreach (string version in values)
+                try
+                {
+                    if (Directory.Exists(pathToTempFolder))
+                    {
+                        Directory.Delete(pathToTempFolder, true);
```

Comment:
Why are you calling delete each time in the loop? Do keep in mind the more often we call Delete the more likely we are going to fail, as anytime a file still has an open handle it will fail this Delete call.

Reply:
but we do need to ensure the old index.json is deleted if there is one present, is there a better way to address a file handle in use?

Reply:
You are deleting the entire directory. It would be better to try and just delete or rename the file if it exists.

Reply:
Assuming your directory is shared. 

Another thing we should probably do (assuming we stick with this) is to set the IndexDirectory to make sure it is shared on the machine and is scoped to the repo.